src/uct/ib/mlx5/rc/rc_mlx5_common.c

```diff
@@ -620,40 +618,53 @@ uct_rc_mlx5_dp_ordering_ooo_init(uct_rc_mlx5_iface_common_t *iface,
      * - if dp_ordering_ooo is set, QPC/DCTC can enable AR.
      * - if dp_ordering_ooo_force is set, QPC/DCTC can request mlxreg
      *   configuration override, useful to force disable.
-     *
-     * QP modify behavior with returned values:
-     * - UCS_AUTO: Do not affect existing system behavior.
-     * - UCS_NO  : Force AR disabling on the QP if supported. QP modify will
-     *   return error on failure.
-     * - UCS_TRY : Set AR to enable, ignored if any failure.
-     * - UCS_YES : Force AR enabling on the QP if supported. QP modify will
-     *   return error on failure.
+     * 
+     * Fail at the following cases:
+     * - Want to force configuration but can't force it
+     * - Want to force enable or force disable of adaptive routing but its unavailable
+     * - Want to force disable adaptive routing but force enabling DDP
      */
 
-    if ((config->super.ar_enable == UCS_TRY) && dp_ordering_ooo) {
-        iface->super.super.config.dp_ordering_ooo = UCS_TRY;
-    } else if (config->super.ar_enable == UCS_NO) {
-        if (!dp_ordering_ooo_force) {
-            goto failure;
-        }
-
-        iface->super.super.config.dp_ordering_ooo = UCS_NO;
-    } else if (config->super.ar_enable == UCS_YES) {
-        if (!dp_ordering_ooo_force || !dp_ordering_ooo) {
-            goto failure;
-        }
+    iface->config.dp_ordering = UCT_IB_MLX5_DP_ORDERING_IBTA;
+
+    min_forced_ordering = (config->ddp_enable == UCS_YES) ?
```

Comment:
requested_ordering maybe

Reply:
```
if (ddp_enable == YES) {
  min_forced_ordering = OOO_ALL;
} else if (ar_enable YES or TRY) {
  min_forced_ordering = OOO_RW;
} else {
  min_forced_ordering = IBTA:
}
```
and compare min_forced ordering with ordering_cap

Reply:
and maybe put sanitisation before.

Reply:
Min forced ordering is being used in the sanitization:
![image](https://github.com/user-attachments/assets/ff6ff37e-22df-473a-879c-800104106fcc)

I think that name is less ambiguous as `requested_ordering` might be a `try`, and the condition above shouldn't fail if we set DDP=try or AR=try and they are not available

Reply:
ok for naming